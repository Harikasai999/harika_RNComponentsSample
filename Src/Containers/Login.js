/**
 * Sample React Native App
 * https://github.com/facebook/react-native
 * @flow
 */

import React, { Component } from 'react'
import { Platform, StyleSheet, Text, View, TouchableOpacity, Dimensions } from 'react-native'
import LinearGradient from 'react-native-linear-gradient'
import { TextField } from 'react-native-material-textfield'
import { Actions } from 'react-native-router-flux'
import CustomNavbar from './../Components/CustomNavbar'
import TextInputField from './../Components/TextInputField'
type Props = {}
const height = Dimensions.get('window').height
const width = Dimensions.get('window').width
var array = [
  {
    state_value: '',
    label_name: 'Email'
  },
  {
    state_value: '',
    label_name: 'Password'
  }
]
class Login extends Component<Props> {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      password: ''
    }
  }
  onChangeText(text) {
    alert('' + text)
    // this.setState({
    // email:
    // })
  }

  render() {
    var label = 'Email'
    return (
      <View style={{ flex: 1 }}>
        <LinearGradient colors={['#243b55', '#141e30', '#243b55']} start={{ x: 0.0, y: 0.5 }} end={{ x: 1.0, y: 0.5 }} locations={[0.0, 1.0]} style={styles.container}>
          <View style={{ height: 44 }}>
            <CustomNavbar title="Login" />
          </View>
          <View styles={styles.container}>
            {array.map((res, key) => {
              return <TextInputField stateValue={res.state_value} Label={res.label_name} onChangeFunc={this.onChangeText.bind(this)} reference={res.label_name} />
            })}
          </View>
        </LinearGradient>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  subContainer: {
    flex: 1,

    paddingRight: 30,
    paddingLeft: 30
  },
  buttonText: {
    fontSize: 16,
    textAlign: 'center',
    fontWeight: '500',
    color: 'white',
    padding: 10,
    backgroundColor: 'transparent'
  },
  headerText: {
    fontSize: 25,
    textAlign: 'center',
    fontWeight: '800',
    color: 'white',
    marginTop: 50,
    backgroundColor: 'transparent'
  },
  buttonView: {
    marginTop: 15,

    borderRadius: 5,
    borderColor: 'white',
    borderWidth: 1
  },
  viewBorder: {
    height: 500,
    width: 300,
    borderRadius: 10,
    borderColor: 'white',
    borderWidth: 2
  },

  labelTextStyle: {
    padding: 5
  },
  linearGradient: {
    marginTop: 10,
    padding: 10,
    borderRadius: 5,
    borderWidth: 2,
    borderColor: 'white'
  }
})
export default Login
