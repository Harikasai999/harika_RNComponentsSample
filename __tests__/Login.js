import 'react-native'
import React from 'react'
import Login from '../Src/Login'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const login = renderer.create(<Login />).toJSON()
  expect(login).toMatchSnapshot()
})
