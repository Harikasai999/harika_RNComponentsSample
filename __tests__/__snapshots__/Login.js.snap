// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <BVLinearGradient
    colors={
      Array [
        4290415362,
        4279769112,
      ]
    }
    locations={null}
    style={
      Object {
        "flex": 1,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "white",
          "fontSize": 25,
          "fontWeight": "800",
          "marginTop": 50,
          "textAlign": "center",
        }
      }
    >
      Login
    </Text>
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "height": 64,
            "padding": 5,
            "paddingBottom": 8,
            "paddingTop": 32,
          }
        }
      >
        <View
          style={
            Object {
              "position": "absolute",
              "top": 36,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 16,
                "padding": 5,
              }
            }
          >
            Email
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            disableFullscreenUI={true}
            editable={true}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            selectionColor="white"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 0,
                  "padding": 0,
                  "top": 2,
                },
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "height": 24,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 8,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "bottom": 0,
                "left": 0,
                "paddingVertical": 4,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgb(213, 0, 0)",
                  "fontSize": 0,
                  "opacity": 0,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "bottom": 0,
                "left": 0,
                "paddingVertical": 4,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 12,
                  "opacity": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      onResponderRelease={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="auto"
    >
      <View
        style={
          Object {
            "backgroundColor": "transparent",
            "borderBottomColor": "rgba(255, 255, 255, 1)",
            "borderBottomWidth": 1,
            "height": 64,
            "padding": 5,
            "paddingBottom": 8,
            "paddingTop": 32,
          }
        }
      >
        <View
          style={
            Object {
              "position": "absolute",
              "top": 36,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "color": "rgba(255, 255, 255, 1)",
                "fontSize": 16,
                "padding": 5,
              }
            }
          >
            Password
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="sentences"
            disableFullscreenUI={true}
            editable={true}
            onBlur={[Function]}
            onChange={[Function]}
            onChangeText={[Function]}
            onContentSizeChange={[Function]}
            onFocus={[Function]}
            selectionColor="white"
            style={
              Array [
                Object {
                  "flex": 1,
                  "margin": 0,
                  "padding": 0,
                  "top": 2,
                },
                Object {
                  "color": "white",
                  "fontSize": 16,
                  "height": 24,
                  "textAlign": "left",
                },
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 8,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "bottom": 0,
                "left": 0,
                "paddingVertical": 4,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "rgb(213, 0, 0)",
                  "fontSize": 0,
                  "opacity": 0,
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
                "bottom": 0,
                "left": 0,
                "paddingVertical": 4,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "white",
                  "fontSize": 12,
                  "opacity": 1,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "white",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "white",
            "fontSize": 16,
            "fontWeight": "500",
            "padding": 10,
            "textAlign": "center",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "transparent",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "500",
              "padding": 10,
              "textAlign": "center",
            },
            Object {
              "fontWeight": "700",
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Sign Up
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "transparent",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginTop": 15,
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 16,
              "fontWeight": "500",
              "padding": 10,
              "textAlign": "center",
            },
            Object {
              "fontWeight": "700",
              "textDecorationLine": "underline",
            },
          ]
        }
      >
        Back
      </Text>
    </View>
  </BVLinearGradient>
</View>
`;
